ground_truth_labeling_jobs/ground_truth_conversion_scripts/GT_semantic_segmentation_to_COCO.ipynb

```diff
@@ -0,0 +1,405 @@
+    "    annotation = {\n",
+    "            \"category_id\": category_id,\n",
+    "            \"id\": idd,\n",
+    "            \"image_id\": image_id,\n",
+    "            \"iscrowd\": 0,\n",
+    "            \"segmentation\": [],\n",
+    "            \"area\": area_GT.tolist(),\n",
+    "            \"bbox\": bb_GT.tolist(),\n",
+    "        }\n",
+    "    for contour in contours:\n",
+    "        contour = np.flip(contour, axis=1)\n",
+    "        segmentation = contour.ravel().tolist()\n",
+    "        annotation[\"segmentation\"].append(segmentation)\n",
+    "    return annotation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download the mask output from S3\n",
+    "\n",
+    "### WARNING: This cell will download the annotated masks from S3 bucket.\n",
+    "\n",
+    "If you have a very large dataset, you can either generate the segmentation data from masks using batches or make sure you have large enough storage to cater for the downloaded data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import time\n",
+    "annotations = []\n",
+    "path = \"/\".join(data_objs[0][JOBNAME+'-ref'].split(\"/\")[:-1]) + \"/\"\n",
+    "!aws s3 cp --recursive $path ./output_mask/"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate segmentation data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import matplotlib\n",
+    "from skimage import io\n",
+    "import matplotlib.pyplot as plt\n",
+    "from skimage.color import rgb2gray\n",
+    "\n",
+    "idd = 0\n",
+    "all_annotations = []\n",
+    "annotations = []\n",
+    "for image_id, file in enumerate(data_objs):\n",
+    "    segments = []\n",
+    "    img_file = os.path.join('output_mask/', os.path.basename(file[JOBNAME + '-ref']))\n",
+    "    a = io.imread(img_file,plugin='matplotlib')\n",
+    "    colors = np.unique(a.reshape(-1, a.shape[2]), axis=0)\n",
+    "    for i in range(colors.shape[0]):\n",
+    "        if colors[i][:-1].mean() < 1:\n",
+    "            segments.append(colors[i])\n",
+    "    \n",
+    "    for idds, seg in enumerate(range(len(segments))):\n",
+    "        color_hex = matplotlib.colors.to_hex(segments[seg])\n",
+    "        category_id = mapping[color_hex]\n",
+    "        img = io.imread(img_file, plugin='matplotlib')\n",
+    "        msk = segments[seg]  \n",
+    "        masked_img = cv2.inRange(img, msk, msk)\n",
+    "        if len(segments) > 1:\n",
+    "            img[masked_img>0]=(255,255,255)\n",
+    "        annot = generate_segmentation(img,category_id, idd, image_id)\n",
+    "        idd+=1\n",
+    "        annotations.append(annot)\n",
+    "        print(\"Processing image {}:\".format())"
```

Comment:
I think this is missing the image index, and maybe it should go at the top of the loop?

Reply:
Fixed